src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,234 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
+    }
+
+    virtual void TearDown(void)
+    {
+        circular_buffer_destroy(circular_buffer);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
+class CircularBufferTestInit0 : public testing::Test
+{
```

Comment:
Don't need to set up a whole class to test the constructor. Use TEST instead of TEST_F and write whatever test code

Reply:
Done!